File: VB/ValidationProviderSuccessEvent/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.DXErrorProvider

Namespace ValidationProviderSuccessEvent
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim prov As New MyValidationProvider()
			prov.ValidationMode = ValidationMode.Auto
			AddHandler prov.ValidationSuccessfullyComplete, AddressOf OnValidationSuccessfullyComplete

			Dim notEmptyValidationRule As New ConditionValidationRule()
			notEmptyValidationRule.ConditionOperator = ConditionOperator.IsNotBlank
			notEmptyValidationRule.ErrorText = "Please enter some text here"
			notEmptyValidationRule.ErrorType = ErrorType.Warning

			prov.SetValidationRule(textEdit1, notEmptyValidationRule)
			prov.SetValidationRule(textEdit2, notEmptyValidationRule)
			prov.SetValidationRule(textEdit3, notEmptyValidationRule)

			ValidateChildren()
		End Sub

		Private Sub OnValidationSuccessfullyComplete(ByVal sender As Object, ByVal e As EventArgs)
			simpleButton1.Enabled = True
			simpleButton1.Focus()
		End Sub
	End Class
End Namespace

File: CS/ValidationProviderSuccessEvent/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraEditors.DXErrorProvider;

namespace ValidationProviderSuccessEvent
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            MyValidationProvider prov = new MyValidationProvider();
            prov.ValidationMode = ValidationMode.Auto;
            prov.ValidationSuccessfullyComplete += OnValidationSuccessfullyComplete;

            ConditionValidationRule notEmptyValidationRule = new ConditionValidationRule();
            notEmptyValidationRule.ConditionOperator = ConditionOperator.IsNotBlank;
            notEmptyValidationRule.ErrorText = "Please enter some text here";
            notEmptyValidationRule.ErrorType = ErrorType.Warning;

            prov.SetValidationRule(textEdit1, notEmptyValidationRule);
            prov.SetValidationRule(textEdit2, notEmptyValidationRule);
            prov.SetValidationRule(textEdit3, notEmptyValidationRule);

            ValidateChildren();
        }

        private void OnValidationSuccessfullyComplete(object sender, EventArgs e)
        {
            simpleButton1.Enabled = true;
            simpleButton1.Focus();
        }
    }
}
